### ISSUE TITLE
Investigate AWS Data Migration Service (DMS) for migrating services' databases

### ISSUE THREAD

**Issue by digitalronin** [ministryofjustice/cloud-platform#924]
AWS DMS claims to offer the ability to migrate databases across AWS accounts without downtime (although older DBs have to be restarted after setting a flag, which will cause a brief outage).

Other teams have experienced data loss (see [this ADR](https://github.com/ministryofjustice/laa-hosting-architectural-decisions/blob/master/doc/adr/0008-oracle-data-migration.md), using an older version of DMS, so although it looks promising, we need to be thorough and cautious to confirm that it is a good fit for our needs.

This ticket is a placeholder, and the tasks listed may need to be broken down into smaller tickets.

### Tasks

 - [x] Establish how much can be scripted, and how - CLI or Terraform?
 - [x] Create a checklist for teams, listing tasks they need to do
 - [x] Do a realistic test, migrating a production-sized database (4G or so?)
 - [x] Test that target data matches the source, exactly (content, relationships, stored procedures, triggers)
 - [x] Do a load-test, writing & updating the source. Confirm all changes are replicated correctly
 - [x] Create a list of the pre-requisite tasks that CP have to do, to enable a migration
 - [x] Decide if and when DMS is suitable for migrating live services
 - [x] Update the user guide, explaining what DMS is and how to use it to migrate a live service

**Comment by mtrblmoj** [comment]
Notes - 
DMS is suitable if (or at least tested for) : 
- Going from postgres 9.4.9+ to 9.4.9+ (ideally 9.6+)
- If the Cloud-Platform team has access to the AWS account hosting the source database.
- If the team has access to a superadmin user on the source and target RDS, 
- If the target DB can be safely wiped.
- **if the source database can be restarted a couple of time, for DMS prep**

**Comment by mtrblmoj** [comment]
Notes - 
DMS pre-requirements :
- Source RDS is made publicly available (locked down of course)
- Source Security Groups is allowing input from the cluster (live-1 igw)
- Source RDS parameter group is modified according to AWS support ticket
- Source RDS is restarted (2 min outage)
- post-migration validation defined
- post-migration rollback defined

**Comment by mtrblmoj** [comment]
Notes -
We ran a DMS migration for PVB staging, we faced the following issues : 
- DMS failed when the tables it was trying to migrate were already on target. We `DROP .. CASCADE`
- DMS failed to migrate from PVB's 9.4.7 database. We upgraded to 9.4.9, using the _modify_ feature on RDS
- DMS failed to process large json fields. The migration settings had to be changed to :
```FullLobMode": false,
    "LobChunkSize": 64,
    "LimitedSizeLobMode": true,
    "LobMaxSize": 20,
```
- Secondary objects, such as foreign keys, indexes have not been migrated.  Currently investigating the best way to do so.
